sync_poltens.py

```diff
@@ -0,0 +1,342 @@
+    pol_mask = hp.read_map(str(DATA_DIR / MAPS["PolMask"]))
+    int_map = hp.read_map(str(DATA_DIR / MAPS["Haslam"])) * 1e6 # map in K, convert to muK
+    int_map *= (23. / 0.408) **  -3 # scale to 23 GHz
+    pol_map = hp.read_map(str(DATA_DIR / MAPS["KBand"]), field=(1, 2)) * 1e3 # map in mK, convert to muK
+    
+    int_mask = hp.ud_grade(int_mask, nside_out=nside)
+    pol_mask = hp.ud_grade(pol_mask, nside_out=nside)
+
+    # combine int and pol masks. This will be necessary as
+    # frolov transform mixes stokes parameters, requiring common
+    # analysis region.
+    joint_mask = np.logical_and(int_mask, pol_mask)
+    for is_Dell in [True, False]:
+        binning = nmt.NmtBin(nside=nside, nlb=1, lmax=lmax, is_Dell=is_Dell)
+        f2 = nmt.NmtField(joint_mask, pol_map)
+        f0 = nmt.NmtField(joint_mask, [int_map])
+
+        cl_22 = nmt.compute_full_master(f2, f2, binning)
+        cl_00 = nmt.compute_full_master(f0, f0, binning)
+
+        cl = np.concatenate([binning.get_effective_ells()[None, :], cl_00, cl_22])
+        if is_Dell:
+            fname = "haslam_kband_spectra_dl.fits"
+        else:
+            fname = "haslam_kband_spectra_cl.fits"
+        hp.write_cl(str(DATA_DIR / fname), cl, overwrite=True)
+
+    iqu = hp.read_map(str(DATA_DIR / "transformed_iqu_fwhm2deg_corrected.fits"), field=(0, 1, 2))
+    for is_Dell in [True, False]:
+        binning = nmt.NmtBin(nside=nside, nlb=1, lmax=lmax, is_Dell=is_Dell)
+        f2 = nmt.NmtField(joint_mask, iqu[1:])
+        f0 = nmt.NmtField(joint_mask, [iqu[0]])
+
+        cl_22 = nmt.compute_full_master(f2, f2, binning)
+        cl_00 = nmt.compute_full_master(f0, f0, binning)
+
+        cl = np.concatenate([binning.get_effective_ells()[None, :], cl_00, cl_22])
+        if is_Dell:
+            fname = "frolov_2degfwhm_corr_dl.fits"
+        else:
+            fname = "frolov_2degfwhm_corr_cl.fits"
+        hp.write_cl(str(DATA_DIR / fname), cl, overwrite=True)
+
+    return 
+
+def PlotPowerspectra():
+    """ Plot the powerspectra of the Haslam and WMAP Kband maps, scaled
+    to 23 GHz, and plotted in uK^2.
+    """
+    dl = hp.read_cl(str(DATA_DIR / "haslam_kband_spectra_dl.fits"))
+    fig, ax = plt.subplots(1, 1)
+    ax.loglog(dl[0], dl[1], label=r"${\rm TT,~Haslam}$")    
+    ax.loglog(dl[0], dl[2], label=r"${\rm EE,~K~band}$")
+    ax.loglog(dl[0], dl[5], label=r"${\rm BB,~K~band}$")
+    ax.spines['top'].set_visible(False)
+    ax.spines['right'].set_visible(False)
+    ax.set_xlabel(r"$\ell$")
+    ax.set_ylabel(r"$\ell (\ell + 1) / 2 \pi~C_\ell~({\rm \mu K}^2)$")
+    ax.legend(frameon=False)
+    ax.tick_params(direction="inout", which="both")
+    ax.set_title(r"${\rm Data~at~23~GHz}$")
+    fig.savefig("spectra_TTEEBB.pdf")
+
+    dl = hp.read_cl(str(DATA_DIR / "frolov_2degfwhm_corr_dl.fits"))
+    fig, ax = plt.subplots(1, 1)
+    ax.loglog(dl[0], dl[1], label=r"${\rm tt}$")    
+    ax.loglog(dl[0], dl[2], label=r"${\rm ee}$")
```

Comment:
Frolov `iqu` aren't already logs? shouldn't you plot them with `semilogx`?

Reply:
it's not a log quantity, it is the C_ell of the log of the map, how is that related to the original C_ell?

Reply:
Yes you're right, I was just quickly copying a plot and taking a look at this quantity, the ylabel shouldn't be C_ell. It's C_ell of a log map, as you point out.

Reply:
actually, it seems like if you take a log of a map of a powerlaw, it is still a powerlaw, I was confused about map-domain versus spectra. https://gist.github.com/zonca/85f50df02f6e731bfb38240535a8fda8